stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -232,24 +232,37 @@ internal class ChannelControllerImpl(
     }
 
     fun markRead(): Result<Boolean> {
-        if (!getConfig().isReadEvents) return Result(false, null)
+        if (!getConfig().isReadEvents) {
+            return Result(false, null)
+        }
+
         // throttle the mark read
         val messages = sortedMessages()
-        if (messages.isNotEmpty()) {
-            val last = messages.last()
-            val lastMessageDate = last.createdAt ?: last.createdLocallyAt
 
-            if (lastMarkReadEvent == null || lastMessageDate!!.after(lastMarkReadEvent)) {
-                lastMarkReadEvent = lastMessageDate
-                val userRead = ChannelUserRead(domainImpl.currentUser).apply {
-                    lastRead = last.createdAt ?: last.createdLocallyAt
-                }
-                _read.postValue(userRead)
-                client.markMessageRead(channelType, channelId, last.id).execute()
-                return Result(true, null)
-            }
+        if (messages.isEmpty()) {
+            logger.logI("No messages; nothing to mark read.")
+            return Result(false, null)
         }
-        return Result(false, null)
+
+        val last = messages.last()
+        val lastMessageDate = last.let { it.createdAt ?: it.createdLocallyAt }
```

Comment:
@tbarbugli how does the backend API actually handle the mark read call? The usage of createdLocallyAt is a bit funky here

Reply:
`createdLocallyAt` would be used only for offline messages that has not been sent to the server (because no connection, for example)